pkg/operator/ceph/cluster/mon/util.go

```diff
@@ -276,3 +287,70 @@ func fullNameToIndex(name string) (int, error) {
 	}
 	return id, nil
 }
+
+func monConfigString(mons []*monConfig) string {
+	monAddr := ""
+	for _, m := range mons {
+		a := fmt.Sprintf("%s:%d", m.PublicIP, m.Port)
+		if len(monAddr) > 0 {
+			monAddr = monAddr + "," + a
+		} else {
+			monAddr = a
+		}
+	}
+	return monAddr
+}
+
+func monInfoString(mons map[string]*cephconfig.MonInfo) string {
+	monAddr := ""
+	for _, m := range mons {
+		a := m.Endpoint
+		if len(monAddr) > 0 {
+			monAddr = monAddr + "," + a
+		} else {
+			monAddr = a
+		}
+	}
+	return monAddr
+
+}
+
+// ceph csi storage class can embed mon service in secret.
+// list all such storage classes and update their mon values
+func updateMonValuesForSC(context *clusterd.Context, clusterName string, monAddr string) error {
+	listOpts := metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", csiSCLabelSelector, clusterName)}
+	scs, err := context.Clientset.Storage().StorageClasses().List(listOpts)
+	if err != nil {
+		return fmt.Errorf("failed to list storage classes: %+v", err)
+	}
+	var lastErr error
+	for _, sc := range scs.Items {
+		if csiProvisioners.Contains(sc.Provisioner) {
+			logger.Debugf("checking storage class %s", sc.Name)
+			if monValFrom, ok := sc.Parameters[csiSCMonParamName]; ok {
+				// this sc supports mon failover
+				secretNS, ok := sc.Parameters[csiSCSecretNSParamName]
+				if !ok {
+					continue
+				}
+				secretName, ok := sc.Parameters[csiSCSecretParamName]
+				if !ok {
+					continue
+				}
+				secret, err := context.Clientset.Core().Secrets(secretNS).Get(secretName, metav1.GetOptions{})
```

Comment:
it seems odd that a StorageClass would have a namespace-specific resource. Can we expect the namespace to be the same as the namespace where rook is running? Then it will also be more reasonable for the operator to have permissions to the resource in the rook namespace.

Reply:
yes, the secret is better to be in a dedicated namespace that rook also has access.